util/design/gen-lc-state-enc.py

```diff
@@ -12,33 +12,60 @@
 from pathlib import Path
 
 import hjson
+from mako.template import Template
+
 from lib.common import wrapped_docstring
 from lib.LcStEnc import LcStEnc
-from mako.template import Template
 
 # State encoding definition
 LC_STATE_DEFINITION_FILE = "hw/ip/lc_ctrl/data/lc_ctrl_state.hjson"
 # Code templates to render
 TEMPLATES = ["hw/ip/lc_ctrl/rtl/lc_ctrl_state_pkg.sv.tpl"]
 
 
+def _render_template(template: str, output_file: str, lc_st_enc: LcStEnc):
+    with open(template, 'r') as tplfile:
+        tpl = Template(tplfile.read())
+        if output_file:
+            with open(output_file, 'wb', buffering=2097152) as outfile:
```

Comment:
Can you add a comment describing why the `buffering` attribute is needed? The value seems a bit magic to me.

Reply:
Done.

Not sure why that particular size. It was added in #17574 by @mundaym. Looks like it is 2^21.